Documentation/html_viewer/js/dictionary_functions.js

```diff
@@ -85,20 +85,24 @@
    		var tokensOfTerm = tTerm.match(/\S+/g);
 		
 		
-		// Generate two regexes - one that matches all syntax that 
+		// Generate three regexes - one that matches all syntax that 
 		// contains each search term, and one that matches all syntax that
 		// contains a word beginning with each search term. This way we 
-		// can prioritise matches that start with the search terms.
-    	var matchExp = '';
-    	var priorityMatch = '';
-    	$.each(tokensOfTerm, function(index, matchToken)
-    	{
-       		matchExp += '(?=.*' + matchToken + ')';
-       		priorityMatch += '(?=.*\\b' + matchToken + ')';
-   		});
+		// can prioritize matches that start with the search terms.
+        // Third regex will match full words.
+        var matchExp = '';
+        var priorityMatch = '';
+        var wordMatch = '';
+        $.each(tokensOfTerm, function(index, matchToken)
+        {
+            matchExp += '(?=.*' + matchToken + ')';
+            priorityMatch += '(?=.*\\b' + matchToken + ')';
+            wordMatch += '(?=.*\\b' + matchToken + '\\b)';
```

Comment:
@bwmilby @livecodeali I'm wondering if this should be tweaked so that the sort order is:

1. whole match of display name
2. display name begins with token
3. display name contains token
4. syntax contains token somewhere

Reply:
That would be a very easy thing to add.  I'm not sure if we should add 3 or replace 2 with 3.  I'm also wondering how this is impacting the speed of the sort.  It seems faster, but I want to do some actual tests and compare the original to the update (could even do a/b/c with above as c).  I need to figure out the best way to do that - probably with a stand alone browser.

Reply:
I think we might need to keep 3. because of LiveCode Builder's habit of having display names that are distinct from the syntax.